avocado/plugins/spawners/podman.py

```diff
@@ -129,3 +132,55 @@ async def wait_task(task_info):
             if not PodmanSpawner.is_task_alive(task_info):
                 return
             await asyncio.sleep(0.1)
+
+    @staticmethod
+    async def _check_requirement_core_avocado():
+        env_type = 'podman'
+        env = settings.as_dict().get('spawner.podman.image')
+        req_type = 'core'
+        req = 'avocado'
+
+        on_cache = get_requirement_on_cache(env_type, env,
+                                            req_type, req)
+        if on_cache:
+            return True
+
+        podman_bin = settings.as_dict().get('spawner.podman.podman_bin')
+        try:
+            # pylint: disable=E1133
+            proc = await asyncio.create_subprocess_exec(
+                podman_bin, "run", env,
+                "avocado", "magic",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE)
+        except (FileNotFoundError, PermissionError):
+            return False
+
+        await proc.wait()
+        if proc.returncode != 0:
+            return False
+
+        stdout = await proc.stdout.read()
+        result = stdout.rstrip().decode() == MAGIC
+        if result:
+            set_requirement_on_cache(env_type, env,
+                                     req_type, req)
+            return True
+        return False
+
+    @staticmethod
+    async def check_task_requirements(task_info):
+        runnable_requirements = task_info.task.runnable.requirements
```

Comment:
Here is one example where task and task_info are combined. You are already passing the task obj inside the task_info. So this could be `task.runnable.requirements` ;)

Reply:
Right, the composition is intentional.  The requirements check may also add to the `task_info.status` about what happened there.